app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Admin
+  class BaseController < ApplicationController
+    before_action :validate_admin
+
+    private
+
+    def validate_admin
+      authorize(current_user, :admin?)
```

Comment:
Since we have pundit I've decided to use it here

Reply:
I think the point of the application policy is to allow us to just have an `authorize` call in each of the admin controllers, without having to create policies for each resource (or maybe we need to, but each of them could just be empty and extending `ApplicationPolicy`).

I'm not sure if I would like to do that instead of what you do here. This has the benefit of keeping the controllers more simple.

Reply:
I would normally authorize each action in controller, however in this case https://github.com/abtion/muffi/blob/main/app/policies/application_policy.rb is set up to check for `user.admin?` in the base level and is not really used, so I'm not sure how to proceed we also only have one "role".

Reply:
I've updated the PR to authorize on each action - except for index which uses policy scope.
That seems to be the intended usage of pundit judging from their documentation.